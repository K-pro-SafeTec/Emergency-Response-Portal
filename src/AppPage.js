import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';

export default ({ title, Component }) => (
  <React.Fragment>
    <AppBar position="static" color="default">
      <Toolbar>
        <Typography variant="title" color="inherit">{title}</Typography>
      </Toolbar>
    </AppBar>
    <Component />
  </React.Fragment>
);
