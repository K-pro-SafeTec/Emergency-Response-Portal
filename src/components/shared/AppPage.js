import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import ArrowBack from '@material-ui/icons/ArrowBack';
import Home from '@material-ui/icons/Home';
import SafetecLogo from '../../icons/safetek_for_now.svg';
import { withStyles } from '@material-ui/core';
import { Link } from 'react-router-dom';

const styles = {
  grow: {
    flexGrow: '1',
  },
  homeButton: {
    marginLeft: '-12px',
  },
  backButton: {
    marginRight: '16px',
  },
};

const AppPage = ({ classes, title, back, children }) => (
  <React.Fragment>
    <AppBar position="static" color="default">
      <Toolbar>
        {back && (
          <div>
            <IconButton className={classes.homeButton} component={Link} to="/">
              <Home />
            </IconButton>
            <IconButton className={classes.backButton} component={Link} to={back}>
              <ArrowBack />
            </IconButton>
          </div>
        )}
        <Typography variant="title" color="inherit" className={classes.grow}>
          {title}
        </Typography>
        <a href="https://www.safetec.no/">
          <img
            src={SafetecLogo}
            alt="Levert av Safetec"
            height="40px"
          />
        </a>
      </Toolbar>
    </AppBar>
    {children}
  </React.Fragment>
);

export default withStyles(styles)(AppPage);
