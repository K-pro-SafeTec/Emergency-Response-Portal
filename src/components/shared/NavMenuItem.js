import React from 'react';
import { Link } from 'react-router-dom';
import Button from '@material-ui/core/Button';
import { withStyles } from '@material-ui/core/styles';

const styles = {
  navigationButton: {
    padding: '1em',
    margin: '1em',
    height: '15em',
    flex: '1 0 20em',
    display: 'block',
    textAlign: 'center',
  },
  icon: {
    width: '10em',
    height: '10em',
  },
};

const NavMenuItem = ({ classes, name, icon, to, ...rest }) => (
  <Button component={Link} to={to} size="large" variant="outlined" className={classes.navigationButton} {...rest}>
    <img src={icon} alt={name} className={classes.icon} />
    <div>
      {name}
    </div>
  </Button>
);

export default withStyles(styles)(NavMenuItem);
