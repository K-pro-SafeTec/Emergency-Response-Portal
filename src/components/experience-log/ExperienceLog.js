import React from 'react';
import NavMenu from '../shared/NavMenu';
import NavMenuItem from '../shared/NavMenuItem';
import AppPage from '../shared/AppPage';
import { Switch, Route } from 'react-router-dom';

// Components
import ExerciseReport from './excercise-report/ExerciseReport';
import PerformanceRequirement from './performance-requirement/PerformanceRequirements';
import ReportGenerator from './report-generator/ReportGenerator';
import EmergencyResponseAction from './emergency-response-action/EmergencyResponseAction';
import LessonsLearned from "./lessons-learned/LessonsLearned"

// Icons
import IconExerciseReports from '../../icons/experience-log/exercise-reports.svg';
import IconPerformanceConditions from '../../icons/experience-log/performance-conditions.svg';
import IconLessonsLearned from '../../icons/experience-log/lessons-learned.svg';
import IconReportGenerator from '../../icons/experience-log/report-generator.svg';
import IconEmergencyResponseActions from '../../icons/experience-log/emergency-response-action.svg';
import IconBranchInformation from '../../icons/experience-log/branch-information.svg';



export default ({ match }) => (
  <Switch>
    <Route exact path={match.path}>
      <AppPage title="Erfaringslogg" back="..">
        <NavMenu>
          <NavMenuItem name="Ã˜vingsrapporter" icon={IconExerciseReports} to="exercise-report/" />
          <NavMenuItem name="Ytelseskrav" icon={IconPerformanceConditions} to="performance-requirements/" />
          <NavMenuItem name="Lessons learned" icon={IconLessonsLearned} to="lessons-learned/" />
          <NavMenuItem name="Rapportgenerator" icon={IconReportGenerator} to="report-generator/" />
          <NavMenuItem name="Beredskapsrelaterte aksjoner" icon={IconEmergencyResponseActions} to="emergency-response-action/" />
          <NavMenuItem name="Bransjeinformasjon" icon={IconBranchInformation} to="branch-information/" />
        </NavMenu>
      </AppPage>
    </Route>
    <Route path={`${match.path}exercise-report/`} component={ExerciseReport} />
    <Route path={`${match.path}performance-requirements/`} component={PerformanceRequirement} />
    <Route path={`${match.path}lessons-learned/`} component={LessonsLearned} />
    <Route path={`${match.path}report-generator/`} component={ReportGenerator} />
    <Route path={`${match.path}emergency-response-action/`} component={EmergencyResponseAction} />
  </Switch>
);
