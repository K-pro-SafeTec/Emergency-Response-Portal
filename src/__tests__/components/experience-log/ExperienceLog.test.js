import React from 'react';
import TestRenderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom'
import AppPage from "../../../components/shared/AppPage";
import ExperienceLog from "../../../components/experience-log/ExperienceLog";
import NavMenuItem from "../../../components/shared/NavMenuItem";

let testInstance;

beforeAll(() => {
  const testRenderer = TestRenderer.create(<Router><ExperienceLog match={{params: {personId: 0}}} /></Router>);
  testInstance = testRenderer.root;
});

test('title is correct', () => {
  expect(testInstance.findByType(AppPage).props.title).toBe("Erfaringslogg");
});

test('nav menu contains all items', () => {
  expect(testInstance.findAllByType(NavMenuItem).length).toBe(6);
});
