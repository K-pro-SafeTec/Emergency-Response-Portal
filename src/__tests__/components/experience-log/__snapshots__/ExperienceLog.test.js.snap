// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <header
    className="MuiPaper-root-14 MuiPaper-elevation4-20 MuiAppBar-root-5 MuiAppBar-positionFixed-6 MuiAppBar-colorDefault-11 mui-fixed"
  >
    <div
      className="MuiToolbar-root-41 MuiToolbar-regular-43 MuiToolbar-gutters-42"
    >
      <a
        className="MuiButtonBase-root-51 MuiIconButton-root-45 AppPage-homeButton-2"
        href="/"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiIconButton-label-50"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-54"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-127"
        />
      </a>
      <button
        className="MuiButtonBase-root-51 MuiButtonBase-disabled-52 MuiIconButton-root-45 MuiIconButton-disabled-49"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiIconButton-label-50"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-54"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </span>
      </button>
      <button
        className="MuiButtonBase-root-51 MuiButtonBase-disabled-52 MuiIconButton-root-45 MuiIconButton-disabled-49"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiIconButton-label-50"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-54"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            />
          </svg>
        </span>
      </button>
      <h2
        className="MuiTypography-root-63 MuiTypography-title-69 MuiTypography-colorInherit-92 AppPage-grow-1"
      >
        Erfaringslogg
      </h2>
      <a
        className="AppPage-safetecLink-3"
        href="https://www.safetec.no/"
      >
        <img
          alt="Levert av Safetec"
          height="44px"
          src="safetec.svg"
          width="153px"
        />
      </a>
    </div>
  </header>,
  <div
    className="AppPage-content-4"
  >
    <div
      className="NavMenu-content-98"
    >
      <a
        className="MuiButtonBase-root-51 MuiButton-root-101 MuiButton-outlined-109 MuiButton-sizeLarge-125 NavMenuItem-navigationButton-99"
        href="/exercise-report/"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-102"
        >
          <img
            alt="Øvingsrapporter"
            className="NavMenuItem-icon-100"
            src="exercise-reports.svg"
          />
          <div>
            Øvingsrapporter
          </div>
        </span>
        <span
          className="MuiTouchRipple-root-127"
        />
      </a>
      <a
        className="MuiButtonBase-root-51 MuiButton-root-101 MuiButton-outlined-109 MuiButton-sizeLarge-125 NavMenuItem-navigationButton-99"
        href="/performance-requirements/"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-102"
        >
          <img
            alt="Ytelseskrav"
            className="NavMenuItem-icon-100"
            src="performance-conditions.svg"
          />
          <div>
            Ytelseskrav
          </div>
        </span>
        <span
          className="MuiTouchRipple-root-127"
        />
      </a>
      <a
        className="MuiButtonBase-root-51 MuiButton-root-101 MuiButton-outlined-109 MuiButton-sizeLarge-125 NavMenuItem-navigationButton-99"
        href="/lessons-learned/"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-102"
        >
          <img
            alt="Lessons learned"
            className="NavMenuItem-icon-100"
            src="lessons-learned.svg"
          />
          <div>
            Lessons learned
          </div>
        </span>
        <span
          className="MuiTouchRipple-root-127"
        />
      </a>
      <a
        className="MuiButtonBase-root-51 MuiButton-root-101 MuiButton-outlined-109 MuiButton-sizeLarge-125 NavMenuItem-navigationButton-99"
        href="/report-generator/"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-102"
        >
          <img
            alt="Rapportgenerator"
            className="NavMenuItem-icon-100"
            src="report-generator.svg"
          />
          <div>
            Rapportgenerator
          </div>
        </span>
        <span
          className="MuiTouchRipple-root-127"
        />
      </a>
      <a
        className="MuiButtonBase-root-51 MuiButton-root-101 MuiButton-outlined-109 MuiButton-sizeLarge-125 NavMenuItem-navigationButton-99"
        href="/emergency-response-actions/"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-102"
        >
          <img
            alt="Beredskapsrelaterte aksjoner"
            className="NavMenuItem-icon-100"
            src="emergency-response-action.svg"
          />
          <div>
            Beredskapsrelaterte aksjoner
          </div>
        </span>
        <span
          className="MuiTouchRipple-root-127"
        />
      </a>
      <a
        className="MuiButtonBase-root-51 MuiButton-root-101 MuiButton-outlined-109 MuiButton-sizeLarge-125 NavMenuItem-navigationButton-99"
        href="/industry-information/"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-102"
        >
          <img
            alt="Bransjeinformasjon"
            className="NavMenuItem-icon-100"
            src="industry-information.svg"
          />
          <div>
            Bransjeinformasjon
          </div>
        </span>
        <span
          className="MuiTouchRipple-root-127"
        />
      </a>
    </div>
  </div>,
]
`;
